### PR TITLE
replace Criterion with Divan

### PR DESCRIPTION
**Pull request by FilipAndersson245**
# Description
 
This is a test of changing out the current criterion microbenchmark tool to [Divan](https://nikolaivazquez.com/blog/divan/), a new and more straightforward microbenchmark suit.
Itself states it is robust to noise, and even allow it to be used in CI settings. It by default has no external dependencies and is very fast to run, the sampling method allows it to be a lot faster compared to criterion requiring less samples.
The output is also nicely displayed and easy to get a quick overview of the performance.

image

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by FilipAndersson245** [comment]
I also saw that that [this](https://nikolaivazquez.com/blog/divan/#measure-allocations) exists as well, allowing you to track allocation and deallocations, to easily find whenever a change affects how much memory is being allocated.

**Comment by FilipAndersson245** [comment]
btw wooa nushells 12000 PR

**Comment by FilipAndersson245** [comment]
Anyone in the core team with a thought on this change, i think this is a good step to simplify the benchmark suit, as its bott faster and require you to look on large graph to get a ballpark of the performance, i also think we should expand the test skit adding for exemplet bechmark of how long std takes to eval etc.

**Comment by kubouch** [comment]
I don't have a problem with it. The benchmarks are currently not used frequently because we lack any kind of continuous monitoring.

**Comment by rgwood** [comment]
Agreed with the above. Divan looks nice, Criterion can be difficult to work with and slow. I mostly used it because at the time it was the most popular benchmarking framework for Rust.

I'd be happy to try this out but I'd defer to @amtoine since it will impact his collection of performance data over time.

Thanks for the PR!

**Comment by rgwood** [comment]
Confirmed with Antoine that he's OK with this. It's a little unfortunate to cause a "break" in our performance tracking history, but the timing is such that there isn't much harm.

**Comment by rgwood** [comment]
Can you please update `benches/README.md` as part of this PR? Otherwise it will be out of date.

**Review by rgwood — CHANGES_REQUESTED** [review]
Can you please update benches/README.md as part of this PR? Otherwise it will be out of date.

File: benches/benchmarks.rs
Comment (FilipAndersson245) [reply benches/benchmarks.rs:58]: As i understand this is more a think how benchmarks work in cargo, Divan being more lightweight may reduce the overhead compared to Criterion but I think that can be done later if it fells suitable.

**Comment by FilipAndersson245** [comment]
> Can you please update benches/README.md as part of this PR? Otherwise it will be out of date.

Done

**Review by rgwood — APPROVED** [review]

**Pull request merged**
